src/libs/SidebarUtils.ts

```diff
@@ -286,7 +300,10 @@ function getOptionData(
     result.shouldShowSubscript = ReportUtils.shouldReportShowSubscript(report);
     result.pendingAction = report.pendingFields ? report.pendingFields.addWorkspaceRoom || report.pendingFields.createChat : undefined;
     result.allReportErrors = OptionsListUtils.getAllReportErrors(report, reportActions) as OnyxCommon.Errors;
-    result.brickRoadIndicator = Object.keys(result.allReportErrors ?? {}).length !== 0 ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : '';
+    result.brickRoadIndicator =
+        Object.keys(result.allReportErrors ?? {}).length !== 0 || (canUseViolations && ReportUtils.transactionThreadHasViolations(report, transactionViolations, null, parentReportAction))
+            ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR
+            : '';
```

Comment:
Let's make this more readable
```suggestion
    const hasErrors = Object.keys(result.allReportErrors ?? {}).length !== 0;
    const hasViolations = canUseViolations && ReportUtils.transactionThreadHasViolations(report, transactionViolations, null, parentReportAction);
    result.brickRoadIndicator = hasErrors || hasViolations ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : '';
```